frappe/core/doctype/document_naming_rule/document_naming_rule.py

```diff
@@ -6,8 +6,19 @@
 import frappe
 from frappe.model.document import Document
 from frappe.utils.data import evaluate_filters
+from frappe import _
 
 class DocumentNamingRule(Document):
+	def validate(self):
+		self.validate_fields_in_conditions()
+
+	def validate_fields_in_conditions(self):
+		for condition in self.conditions:
+			docfields = frappe.get_meta(self.document_type).fields
+			matching_field = list(filter(lambda x: x.fieldname == condition.field, docfields))
+			if not len(matching_field):
```

Comment:
Can this be simplified as `if not matching_field`? If not, add a comment above it to state the reasoning.


In fact, I think we can just replace all this into something more simpler...like

```python
docfields = [x.fieldname for x in frappe.get_meta(self.document_type).fields]
for condition in self.conditions:
	if condition.field not in docfields:
		frappe.throw(_("{0} is not a field of doctype {1}").format(frappe.bold(condition.field), frappe.bold(self.document_type
```